src/services/api.tsx

```diff
@@ -19,55 +101,182 @@ const getAge = (birthDate: Date) => {
+      })
+    )
   );
+
+  let finalBundle: Bundle = bundles[0];
+  bundles.map((bundle: Bundle) => {
+    if (finalBundle !== bundle && bundle.entry.length > 0) {
+      let listId = bundle.entry.map((x: any) => x.resource.id);
+      finalBundle.entry = finalBundle.entry.filter(
+        (entry: any) => listId.indexOf(entry.resource.id) >= 0
+      );
+    }
+    return false;
+  });
+  finalBundle.total = finalBundle.entry.length;
+
+  return addPatientsToBundle(finalBundle);
 };
 
-export const getPatientData = async (patientId: string) => {
+export const requestNextPatients = async (bundle: PatientBundle) => {
   /*
-    getPatientData requests for data from Patient resourcce of id patientId
-    return a Patient object.
-  */
-  let response = await client.search({
-    type: "Patient",
-    patient: patientId
+   * function requestNextPatients
+   * Return the same bundle with more patients, fetched from the nextLink attribute.
+   */
+  if (!bundle.nextLink) {
+    console.info("no link available");
+    return;
+  } else {
+    let newBundle: PatientBundle = (await makeRequestByURL(
+      bundle.nextLink
+    )) as PatientBundle;
+
+    bundle.entry = bundle.entry.concat(newBundle.entry);
+
+    newBundle = addPatientsToBundle(newBundle);
+    bundle.patients = bundle.patients.concat(newBundle.patients);
+    bundle.nextLink = newBundle.nextLink;
+    return bundle;
+  }
+};
+
+export const addPatientsToBundle = (bundle: Bundle) => {
+  /*
+   *  addPatientsToBundle transforms a bundle in a PatientBundle (generate Patient objects)
+   */
+
+  let response: PatientBundle = bundle as PatientBundle;
+  response.patients = response.entry.map((entry: any) => {
+    const patient: Patient = {
+      id: entry.resource.id,
+      birthDate: entry.resource.birthDate,
+      age:
+        entry.resource.birthDate && getAge(new Date(entry.resource.birthDate))
+    };
+    if (entry.resource.name) {
+      if (entry.resource.name[0].given)
+        patient.firstName = entry.resource.name[0].given.join(", ");
+      if (entry.resource.name[0].family)
+        patient.lastName = entry.resource.name[0].family;
+    }
+    if (entry.resource.identifier) {
+      patient.identifier = entry.resource.identifier
+        .map((e: any) => {
+          return e.value;
+        })
+        .join(", ");
+    }
+    return patient;
   });
-  if (!response.data.entry) return; //if there is no patient found, response.data.entry is undefined.
-  const patientData = response.data.entry[0];
+
+  return response;
+};
+
+export const getPatientData = async (patientId: string, detailed?: boolean) => {
+  /*
+   * getPatientData requests for data from Patient resource of id patientId. Used to load all data to show all patient hospitalizations.
+   * patientId
+   * detailed: get detailed informations about patient or not ? (slower, more API calls)
+   * return a Patient object.
+   */
+  let response: any = await makeRequest("Patient", false, "_id=" + patientId);
```

Comment:
Use backquotes and `${}` to build strings instead of `+` concatenation

Reply:
done